### PR TITLE
support for gcs file sync using s3 compatible api

### PR DESCRIPTION
**Pull request by nwingert**
Basic copy/paste job from s3 code since GCS API is compatible with S3 API. Modified some config field names to be more generic (eg 'AccessKey' instead of 'AWSAccessKey') which required modifications to other apps in repo. Also had to update some package references to build properly with gomod. GCS file sync was tested against live GCS bucket but s3/gcs unit tests currently failing - looking into this now

### CODE REVIEW

File: file/gcs/read.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
+package gcs
+
+import (
+	"bufio"
+	"compress/gzip"
+	"crypto/md5"
+	"path/filepath"
+
+	"github.com/minio/minio-go"
+
```
Comment (jbsmith7741) [inline file/gcs/read.go]: extra spaces on imports

File: tmpl/tmpl.go [hunk 2]
```diff
@@ -87,6 +89,12 @@ func Parse(s string, t time.Time) string {
 	hour := fmt.Sprintf("%02d", t.Hour())
 	s = regHour.ReplaceAllString(s, hour)
 
+	// {HOST}
+	if h, err := os.Hostname(); err == nil {
+		re := regexp.MustCompile(`(?i){host}`)
```
Comment (jbsmith7741) [inline tmpl/tmpl.go]: regexp.MustCompile is super slow either set the regex as a var 
``` go
var hostRegex := regexp.MustCompile(`(?i){host}`)
```
or just do two strings.Replace 
``` go 
s = strings.Replace(s, "{host}")
s = strings.Replace(s, "{HOST}")
```

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	github.com/robfig/cron v0.0.0-20180505203441-b41be1df6967
+	github.com/smartystreets/goconvey v0.0.0-20190330032615-68dc04aab96a // indirect
+	github.com/stretchr/testify v1.3.0
+	golang.org/x/crypto v0.0.0-20190404164418-38d8ce5564a5 // indirect
+	golang.org/x/net v0.0.0-20190403144856-b630fd6fe46b // indirect
+	google.golang.org/appengine v1.5.0 // indirect
+	gopkg.in/BurntSushi/toml.v0 v0.3.1
+	gopkg.in/dustinevan/chron.v0 v0.0.0
+	gopkg.in/ini.v1 v1.42.0 // indirect
+	gopkg.in/jbsmith7741/uri.v0 v0.2.0
```
Comment (jbsmith7741) [inline go.mod]: We might want to go through the project and replace imports for gopkg.in/jbsmith7741/uri.v0 with github.com/jbsmith7741/uri. I don't think we need the gopkg.in packages with go modules.

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	github.com/jbsmith7741/trial v0.2.1
+	github.com/jbsmith7741/uri v0.2.0
+	github.com/json-iterator/go v1.1.6
+	github.com/lib/pq v1.0.0
+	github.com/minio/minio-go v6.0.14+incompatible
+	github.com/mitchellh/go-homedir v1.1.0 // indirect
+	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
+	github.com/modern-go/reflect2 v1.0.1 // indirect
+	github.com/nsqio/go-nsq v1.0.7
+	github.com/nwingert/task-tools v0.3.0 // indirect
```
Comment (jbsmith7741) [inline go.mod]: change your path to pcelvng so we can removed this dependency

**Comment by nwingert** [comment]
@jbsmith7741 Can you be more specific about AWSAccessKey change breaking compatibility? I'm not sure I understand that concern.

File: tmpl/tmpl.go (hunk 2 above)
Comment (nwingert) [reply tmpl/tmpl.go]: I'm not sure I understand how `var hostRegex := regexp.MustCompile(`(?i){host}`)` is meaningfully different from `re := regexp.MustCompile(`(?i){host}`)`

**Comment by nwingert** [comment]
I think I've hit all the requested changes

**Pull request merged**